File: qc-application/src/main/java/com/eager/questioncloud/application/api/authentication/SocialAuthentication.java
package com.eager.questioncloud.application.api.authentication;

import com.eager.questioncloud.core.domain.user.User;
import lombok.Getter;

@Getter
public class SocialAuthentication {
    private final User user;
    private final String socialAccessToken;

    private SocialAuthentication(User user, String socialAccessToken) {
        this.user = user;
        this.socialAccessToken = socialAccessToken;
    }

    public static SocialAuthentication create(User user, String socialAccessToken) {
        return new SocialAuthentication(user, socialAccessToken);
    }

    public Boolean isRegistered() {
        return this.user != null;
    }
}

File: qc-application/src/main/java/com/eager/questioncloud/application/api/authentication/dto/SocialAuthentication.kt
package com.eager.questioncloud.application.api.authentication.dto

import com.eager.questioncloud.core.domain.user.model.User

class SocialAuthentication(
    val user: User?,
    val socialAccessToken: String
) {
    val isRegistered: Boolean
        get() = this.user != null

    companion object {
        fun create(user: User?, socialAccessToken: String): SocialAuthentication {
            return SocialAuthentication(user, socialAccessToken)
        }
    }
}
